### ISSUE TITLE
Unexpected suggest highlighting ignores whole word match in favor of matching first letter 

### ISSUE THREAD

**Issue by xlboy** [microsoft/vscode#147423]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.66.2
- OS Version: Darwin x64 21.2.0
- TypeScript Version: 4.6.2

Steps to Reproduce:

When using TypeScript untion intelliSense, a strange character matching was found, which failed to match "ok" in {}.

Judging from the rules in the picture, it seems to be the reason for the "{"symbol?

Is this problem caused by the internal lsp-tsserver?

But when I tried it in WebStorm, there was no such problem.

VSCode: ↓
![84811649908099_ pic](https://user-images.githubusercontent.com/63690944/163310041-068cacfa-6359-4b7c-9db3-55bc2d1a9983.jpg)

WebStorm: ↓
![325251649907961_ pic](https://user-images.githubusercontent.com/63690944/163310123-7e26d02f-60be-4c86-8181-fa5dd5c00ac7.jpg)

TypeScript online playground: ↓
[https://www.typescriptlang.org/play?#code/C4TwDgpgBAylC8UDkBbAhgYwBYEsB2EA9AN44AmAvklAD7JkOMMk4COVtyA9gEa+-EuAawqEArgGcIAJ2p0kZCYrLFJM0cKQBuAFA6MXPBOBQMGAFywE3IdqA](https://www.typescriptlang.org/play?#code/C4TwDgpgBAylC8UDkBbAhgYwBYEsB2EA9AN44AmAvklAD7JkOMMk4COVtyA9gEa+-EuAawqEArgGcIAJ2p0kZCYrLFJM0cKQBuAFA6MXPBOBQMGAFywE3IdqA)

**Comment by mjbvz** [comment]
It looks to me like the only issue is the highlighting of the typed text in the suggestion. This highting is done by VS Code, not tsserver 

I've updated the title to reflect this

**Comment by xlboy** [comment]
> 

When I enter a single letter, it doesn't give advice either.

↓
![29181649983592_ pic](https://user-images.githubusercontent.com/63690944/163499644-c20fba25-97a4-437f-8d5b-e30a62194c0f.jpg)

In theory, it should give a suggestion of "machine/ {id}" | "dddddddd/ {iq}"

**Comment by jrieken** [comment]
We don't consider `{` as booster character (unlike `.`, `_`, casing-changes, etc) and therefore the first character or boost character must match

**Comment by xlboy** [comment]
> 

Ok, got it.

May I ask, What is the reason for not using `{` as a boost character?

**Comment by jrieken** [comment]
No special reason, hasn't been asked for

**Comment by jrieken** [comment]
I have pushed a change to also boost after `{`. This will help with https://github.com/microsoft/vscode/issues/147423#issuecomment-1099748356 and other cases. However, not with the original case (`obobobf{ok}/user`). The `o` can match multiple times. Our strategy rewards word start matches over sequential matches, e.g the later a character matches in a word the higher of a penalty it carries. The penalty isn't compensated with just 2 consecutive matches. ATM there are  is now plans to change that. Closing as half fixed and half as-designed
